File: LAB3/src/main/java/numberMaster/Fibonacci.java
package numberMaster;

public class Fibonacci {
    public static int fib(int n) {
        if (n <= 0) throw new IllegalArgumentException("Illegal number!");
        else if ((n == 1) || (n == 2)) return 1;
        else return fib(n - 1) + fib(n - 2);
    }
}

File: LAB7/fibonacci.py
def fib(n):
    if type(n) is not int:
        return None
    elif n < 0:
        return None
    elif n == 0:
        return 0
    elif (n == 1) or (n == 2):
        return 1
    else:
        return fib(n - 1) + fib(n - 2)
